app/main.py

```diff
@@ -1,2 +1,52 @@
+from typing import Any
+
+
 class Dictionary:
-    pass
+    def __init__(
+            self,
+            initial_capacity: int = 16,
+            load_factor: float = 2 / 3
+    ) -> None:
+        self.capacity = initial_capacity
+        self.load_factor = load_factor
+        self.size = 0
+        self.table = [None] * initial_capacity
+
+    def _resize(self) -> None:
+        self.capacity *= 2
+        new_table = [None] * self.capacity
+        for bucket in self.table:
+            if bucket:
+                for node in bucket:
+                    index = self._hash(node[0])
+                    if not new_table[index]:
+                        new_table[index] = []
+                    new_table[index].append(node)
+        self.table = new_table
+
+    def __setitem__(self, key: str, value: Any) -> None:
+        index = self._hash(key)
+        if not self.table[index]:
+            self.table[index] = []
+        for node in self.table[index]:
+            if node[0] == key:
+                node[2] = value
+                return
+        self.table[index].append([key, self._hash(key), value])
```

Comment:
```suggestion
        self.table[index] = [key, self._hash(key), value]
```

Reply:
If I replace this part, in the case of a collision, it will overwrite the existing entry in self.table[index] instead of adding a new one.